src/sdam/server_description.ts

```diff
@@ -256,14 +256,10 @@ export function compareTopologyVersion(lhs?: TopologyVersion, rhs?: TopologyVers
   }
 
   if (lhs.processId.equals(rhs.processId)) {
-    // TODO: handle counters as Longs
-    if (lhs.counter === rhs.counter) {
-      return 0;
-    } else if (lhs.counter < rhs.counter) {
-      return -1;
-    }
-
-    return 1;
+    // tests mock counter as just number, but in a real situation counter should always be a Long
+    const lhsCounter = Long.isLong(lhs.counter) ? lhs.counter : Long.fromNumber(lhs.counter);
+    const rhsCounter = Long.isLong(rhs.counter) ? lhs.counter : Long.fromNumber(rhs.counter);
+    return lhsCounter.compare(rhsCounter);
```

Comment:
When testing the counter property is a number so `===` works as expected. However the counter is supposed to be a Long and `===` as well as `<` will fail when this is the case. So converting here made sense. There might be a larger issue here of internal communications with mongod losing precision since we don't breakout our own set of bson options, thoughts?